### PR TITLE
Nested Categories

### PR DESCRIPTION
**Pull request by MattRice12**

### CODE REVIEW

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
   def index
     render locals: {
-      categories: Category.all
+      categories: Category.all.where(parent_category_id: nil)
     }
   end
-  
+
+  def show
+    category = Category.find_by(parent_category_id: params.fetch(:id))
```
Comment (kate-walters) [inline app/controllers/categories_controller.rb]: [:id] not (:id) - not sure how much this matters but that's how we've always done it

Comment (MattRice12) [inline app/controllers/categories_controller.rb]: (:id) works with .fetch; [:id] works without .fetch.

**Comment by rluck0419** [comment]
Thumbs up

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+    if category
+      render locals: {
+        category: category
+      }
+    else
+      redirect_to notes_path
```
Comment (DaneCarmichael) [inline app/controllers/categories_controller.rb]: Why are we redirecting to notes path?

File: app/views/categories/_form.html.erb [hunk 3]
```diff
@@ -1,13 +1,15 @@
 <%= form_for(category) do |f| %>
+  <div>
+    <%= f.label :name %>
+    <%= f.text_field :name %>
+  </div>
 
-        <% end %>
+  <div>
+    <label for="category_id">Category: </label>
+    <%= f.select :parent_category_id, options_from_collection_for_select(Category.all, :id, :name) %>
```
Comment (DaneCarmichael) [inline app/views/categories/_form.html.erb]: Should this be all categories, or categories without a parent_id?

**Comment by MattRice12** [comment]
Please push this. I would like, and others probably need, the other changes to the category nesting and traversal to work. I don't think it makes sense to wait until the "create" function to work. I'm not going to work on it again until later tonight.

Let's just create another story for getting the create function to work. So far any category you create just goes to the index. This is giving me more problems than I originally thought it would.

**Pull request merged**